corehq/ex-submodules/casexml/apps/phone/utils.py

```diff
@@ -38,8 +41,20 @@ def delete_sync_logs(before_date, limit=1000, num_tries=10):
     raise CouldNotPruneSyncLogs()
 
 
-def delete_sql_synclogs(before_date):
-    SyncLogSQL.objects.filter(date__lt=before_date).delete()
+def prune_sql_synclogs():
+    """
+    Drops all partition tables containing data that's older than 63 days (9 weeks)
+    """
+    oldest_synclog = SyncLogSQL.objects.aggregate(Min('date'))['date__min']
```

Comment:
Is this needed? I think we could just drop the table and not need to look up the synclogs (the IF EXISTS should catch that)

If it is needed, can you post the SQL this generates and the explain that goes with it? I'm not too familiar with how Django does aggregates like this, but I'm concerned this won't perform well.

Reply:
I used this to pick all the eligible partition tables to delete instead of just the partition-table of the week of 90 days ago which will delete only one partition table but keep the older weeks' partition tables. Practically, this query would run once for a background task.

In terms of performance, since the aggregate query is on partitioned column, it is very efficient. Anyway, I wasn't worried about performance as much since this would generate just one query per week.